lib/Onyx.js

```diff
@@ -997,7 +998,11 @@ function merge(key, value) {
                 // don't apply these changes again
                 delete mergeQueue[key];
 
-                return set(key, modifiedData);
+                // We don't wait until the merge is written to storage.
+                // Instead, we immediately notify subscribers and then return the merge promise
+                const mergePromise = OnyxMerge.merge(key, changes, modifiedData);
```

Comment:
Seeing what `OnyxMerge.native` and `OnyxMerge.web` files do, why not just call `Storage.mege` here

e.g.
- make `Storage.merge` provider for native have the interface of `merge(key, changes)`
- make `Storage.merge` for web have the an interface like `merge(key, changes, modifiedData)` 

So we don't have 2 new files for this 
Also OnyxMerge.merge is a bit weird name/concept

Reply:
Am I seeing this correctly that we will only call `OnyxMerge.merge()` with the final `changes`? The way the merge queue thing works is that it will batch them all together and then do a `set()`. Here we are doing a `merge()` not with the complete partial changes, but only the very last merge call?

Reply:
I think i understand what you guys are saying now.

I already simplified `OnyxMerge.merge()` to `Storage.merge()`, as @kidroca suggested btw...

It's definitely wrong, that we only pass the "current" `changes` from a `Onyx.merge()` call to the `OnyxMerge.merge()` function. From my understanding though, this would mean, that it's not the latest, but the first changes that are passed to `OnyxMerge.merge()`, since the function will early-return if there is already a value in the `mergeQueue`, and thus the `OnyxMerge.merge()` call with the "latest" changes will never happen.

I also went for the approach of removing the `mergeQueue` completely. The only problem in this case is, that we need to make sure, that a previous' `Onyx.merge()` promise has already finished. Otherwise, the second `Onyx.merge()` call will simply overwrite the previous one.

In the tests, this would mean to add a lot of `waitForPromisesToResolve()` calls and in production, we would need to adapt to this behaviour too.

Reply:
I created a separate branch with the `mergeQueue` removed [here](https://github.com/chrispader/react-native-onyx/tree/%40chrispader/feat/use-json-patch-for-merge+remove-merge-queue)